### ISSUE TITLE
ResultListener not receives .dead replies if where statement used

### ISSUE THREAD

**Issue by Metori** [GideonLeGrange/mikrotik-java#27]
If I send "/interface/pppoe-server/server/listen return .id,.dead,disabled,service-name,interface,max-mtu,max-mru,mrru,default-profile,authentication" to the Router and set ResultListener to this query then I will receive all listen data (changed comments, new items, .dead items etc).

But if I send "/interface/listen where type=l2tp-in or type=l2tp-out or type=ovpn-in or type=ovpn-out or type=ppp-in or type=ppp-out or type=pppoe-in or type=pppoe-out or type=pptp-in or type=pptp-out or type=sstp-in or type=sstp-out return .id,.dead,.nextid,disabled,dynamic,running,slave,name,type,comment" and set ResultListener then .dead replies is not receiving at all. All other listen replies are okay in this situation.

Sorry for my english.

**Comment by GideonLeGrange** [comment]
I will look into this. There are two possibilities I see here:
- Some problem with my processing of the “or” syntax which causes your command to fail. 
- Some problem with the internal RouterOS API 

Both are real possibilities. I will lab this over the weekend and let you know. 

Kind regards
Gideon

**Comment by GideonLeGrange** [comment]
I'll get back to you, I've got a few things to test.

**Comment by Metori** [comment]
Is there any progress?

**Comment by Metori** [comment]
What about this?

**Comment by GideonLeGrange** [comment]
I've looked into this some more and don't think this is is not a problem with the Java API. Why I say this requires some knowledge about the underlying RouterOS API. 

The RouterOS API is documented here: http://wiki.mikrotik.com/wiki/Manual:API. The Java API builds up commands that conform to the structures defined in that page, send them across the network and then receives data from the network and decodes it into responses.  

So for example, if you construct a command which reads `/interface print where type=ether or type=vlan`, what is actually sent accross the network is something like this:

```
/interface/print
?type=ether
?type=vlan
?#|!
```
So why is this relevant? 

If you look in the section labeled **Query word** in the page I link to above, you'll see that there is a bullet point that says "Currently only print command handles query words." If you put all the above together, what is basically happening here is that the Java API accepts your `listen` command with the `where` clause and send it to RouterOS, but RouterOS doesn't consider it to be valid. What is bad is that RouterOS doesn't seem to generate an error - if it had sent an error (a `!trap` if you want to see how it looks in the protocol) in response to the command, the Java API would have raised an exception. 

So long story short, this is not a bug in the Java API, but a limitation in the underlying RouterOS API. The workaround I suggest you follow is to use the first command you have above and filter out the types you do not want on your side. It's not as efficient, but it's the only option I see. 

Gideon

**Comment by Metori** [comment]
Thanks for investigation.
